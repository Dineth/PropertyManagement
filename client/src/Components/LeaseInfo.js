import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import leaseInfo from './LeaseInfo.less';


export default class LeaseInfo extends React.Component {

    render () {
        return <div id="leaseInfo">
            <legend>Lease Info</legend>
            <Row><Col>ID</Col><Col>{this.props.leaseInfo.id}</Col></Row>
            <Row><Col>Sart date</Col><Col>{this.props.leaseInfo.start_date}</Col></Row>
            <Row><Col>End date</Col><Col>{this.props.leaseInfo.end_date}</Col></Row>
            <Row><Col>Rent</Col><Col>{this.props.leaseInfo.rent}</Col></Row>
            <Row><Col>Freequency</Col><Col>{this.props.leaseInfo.frequency}</Col></Row>
            <Row><Col>Payment day</Col><Col>{this.props.leaseInfo.payment_day}</Col></Row>
        </div>
    }

}
