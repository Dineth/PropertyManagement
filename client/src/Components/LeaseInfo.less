#leaseInfo {
 border: 2px double black;
 padding: 5px;
}
