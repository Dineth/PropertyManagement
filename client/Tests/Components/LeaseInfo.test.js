import React from 'react';
import renderer from 'react-test-renderer';
import LeaseInfo from '../../src/Components/LeaseInfo.js';

describe('LeaseInfo snap shot test', () => {
        it("it should pass the snapshot test", () => {
            var obj = {
              tenants: [],
              leaseInfo : {
                  id: 123,
                  start_date: '2018-07-12',
                  end_date: '2018-11-17',
                  rent: 545,
                  frequency: 'weekly',
                  payment_day: 'wednesday' },
              leaseData: []
            };

            const component = renderer.create(
              <LeaseInfo leaseInfo={obj.leaseInfo} />,
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

});
