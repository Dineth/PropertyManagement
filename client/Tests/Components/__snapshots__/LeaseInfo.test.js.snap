// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaseInfo snap shot test it should pass the snapshot test 1`] = `
<div
  id="leaseInfo"
>
  <legend>
    Lease Info
  </legend>
  <div
    className="row"
  >
    <div
      className="col"
    >
      ID
    </div>
    <div
      className="col"
    >
      123
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Sart date
    </div>
    <div
      className="col"
    >
      2018-07-12
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      End date
    </div>
    <div
      className="col"
    >
      2018-11-17
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Rent
    </div>
    <div
      className="col"
    >
      545
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Freequency
    </div>
    <div
      className="col"
    >
      weekly
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      Payment day
    </div>
    <div
      className="col"
    >
      wednesday
    </div>
  </div>
</div>
`;
